src/inmanta/data/__init__.py

```diff
@@ -2298,13 +2633,148 @@ async def get_resources_for_version_raw(
         return resources
 
     @classmethod
-    async def get_latest_version(cls, environment: uuid.UUID, resource_id: ResourceIdStr) -> Optional["Resource"]:
+    async def get_latest_version(cls, environment: uuid.UUID, resource_id: m.ResourceIdStr) -> Optional["Resource"]:
         resources = await cls.get_list(
             order_by_column="model", order="DESC", limit=1, environment=environment, resource_id=resource_id
         )
         if len(resources) > 0:
             return resources[0]
 
+    @classmethod
+    def _get_released_resources_base_query(cls):
+        return (
+            f"FROM ( "
+            f"SELECT r.*, row_number() OVER (PARTITION BY r.resource_id ORDER BY r.model DESC) as row_number "
+            f"FROM {cls.table_name()} as r  "
+            f"JOIN public.configurationmodel as cm ON r.model=cm.version "
+            f"WHERE cm.released=TRUE ) as res "
+            f"WHERE row_number = 1 "
+        )
+
+    @classmethod
+    async def get_released_resources(
+        cls,
+        database_order: DatabaseOrder,
+        limit: int,
+        first_id: Optional[str] = None,
+        last_id: Optional[str] = None,
+        start: Optional[Any] = None,
+        end: Optional[Any] = None,
+        connection: Optional[asyncpg.connection.Connection] = None,
+        **query,
+    ) -> List[m.ResourceListElement]:
+        """
+        Get all resources that are in a released version, sorted, paged and filtered
+        """
+        order_by_column = database_order.get_order_by_db_column_name()
+        order = database_order.get_order()
+        filter_statements, values = cls._get_list_query_pagination_parameters(
+            database_order=database_order,
+            id_column=ColumnNameStr("resource_version_id"),
+            first_id=first_id,
+            last_id=last_id,
+            start=start,
+            end=end,
+            **query,
+        )
+
+        db_query = (
+            f"SELECT res.resource_id, res.attributes, res.resource_version_id, res.status, "
+            f"res.resource_type, res.agent, res.value "
+            f"{cls._get_released_resources_base_query()}"
+        )
+        if len(filter_statements) > 0:
+            db_query += f"AND {cls._join_filter_statements(filter_statements)}"
+        backward_paging = ("ASC" in order and end) or ("DESC" in order and start)
+        if backward_paging:
+            if "ASC" in order:
+                backward_paging_order = order.replace("ASC", "DESC")
+            else:
+                backward_paging_order = order.replace("DESC", "ASC")
+
+            db_query += f" ORDER BY {order_by_column} {backward_paging_order}, resource_version_id {backward_paging_order}"
+        else:
+            db_query += f" ORDER BY {order_by_column} {order}, resource_version_id {order}"
+        if limit is not None and limit > 0:
+            db_query += " LIMIT " + str(limit)
```

Comment:
Shouldn't we make `LIMIT` cannot be larger than `DBLIMIT`?

Reply:
You're right, I completely forgot about that